src/core/layout/qgslayoutitemlabel.cpp

```diff
@@ -641,7 +641,31 @@ double QgsLayoutItemLabel::htmlUnitsToLayoutUnits()
 QString QgsLayoutItemLabel::createStylesheet() const
 {
   QString stylesheet;
-  stylesheet += QStringLiteral( "body { margin: %1 %2;" ).arg( std::max( mMarginY * mHtmlUnitsToLayoutUnits, 0.0 ) ).arg( std::max( mMarginX * mHtmlUnitsToLayoutUnits, 0.0 ) );
+  const QString bodyMargin = QStringLiteral( "margin: %1 %2;" ).arg( std::max( mMarginY * mHtmlUnitsToLayoutUnits, 0.0 ) ).arg( std::max( mMarginX * mHtmlUnitsToLayoutUnits, 0.0 ) );
+  QString bodyLineHeight;
+  switch ( mFormat.lineHeightUnit() )
+  {
+    case Qgis::RenderUnit::Percentage:
+      bodyLineHeight += QStringLiteral( "line-height: %1%;" ).arg( mFormat.lineHeight() * 100 );
+      break;
+    case Qgis::RenderUnit::Pixels:
+      bodyLineHeight += QStringLiteral( "line-height: %1px;" ).arg( mFormat.lineHeight() );
+      break;
+    case Qgis::RenderUnit::Points:
+      // While the Qt documentation states this unit type is supported, on Qt 5.15 it is ignored
+      bodyLineHeight += QStringLiteral( "line-height: %1pt;" ).arg( mFormat.lineHeight() );
+      break;
+    case Qgis::RenderUnit::Millimeters:
+      // While the Qt documentation states this unit type is supported, on Qt 5.15 it is ignored
+      bodyLineHeight += QStringLiteral( "line-height: %1cm;" ).arg( mFormat.lineHeight() / 1000 );
+      break;
+    case Qgis::RenderUnit::MetersInMapUnits:
+    case Qgis::RenderUnit::MapUnits:
+    case Qgis::RenderUnit::Inches:
+    case Qgis::RenderUnit::Unknown:
+      break;
+  }
+  stylesheet += QStringLiteral( "body { %1 %2" ).arg( bodyMargin, bodyLineHeight );
 
   QFont f = createDefaultFont();
   stylesheet += QgsFontUtils::asCSS( f, 0.352778 * mHtmlUnitsToLayoutUnits );
```

Comment:
I think the line-height handling should be moved to a QgsTextFormat::asCSS method. This would replace the QgsFontUtils::asCSS call on line 671, and give us a nice logical place to extend the text format -> css conversion capabilities in future (eg handling letter/word spacing)

Reply:
@nyalldawson , done.